File: BOJ/stack/c++/10828.cpp
#include <cstdio>
#include <cstring>

using namespace std;

struct Stack {
    int data[10000];
    int size;
    Stack() {
        size = 0;
    } 
    void push(int num) {
        data[size] = num;
        size += 1;
    }

    int pop() {
        if(empty()) {
            return -1;
        } else {
            size -= 1;
            return data[size];
        }
    }

    bool empty() {
        if(size == 0) {
            return true;
        } else {
            return false;
        }
    }

    int top() {
        if(!empty()) {
            return data[size-1];
        } else {
            return -1;
        }
    }
};

int main (void) {
    int N;
    char cmd[10];
    int num;
    Stack s;
    scanf("%d", &N);

    for(int i = 0; i < N; i++) {
        scanf("%s", cmd);

        if(!strcmp(cmd, "push")) {
            scanf("%d", &num);
            s.push(num);
        } else if(!strcmp(cmd, "top")){
            printf("%d\n",s.top());
        } else if(!strcmp(cmd, "size")) {
            printf("%d\n",s.size);
        } else if(!strcmp(cmd, "pop")) {
            printf("%d\n",s.pop());
        } else {
            if(s.empty()) {
                printf("1\n");
            } else {
                printf("0\n");
            }
        }
    }

    return 0;
}

File: BOJ/stack/java/10828.java
import java.util.Scanner;

class Stack {
    private int[] data = new int[10000];
    private int size;

    public Stack() {
        this.size = 0;
    }

    public void push(int num) {
        this.data[size] = num;
        this.size += 1;
    }

    public int empty() {
        if(this.size == 0) {
            return 1;
        } else {
            return 0;
        }
    }

    public int pop() {
        if(this.empty() == 1) {
            return -1;
        } else {
            this.size -= 1;
            return this.data[this.size];
        }
    }

    public int top() {
        if(this.empty() == 1) {
            return -1;
        } else {
            return this.data[this.size-1];
        }
    }

    public int size() {
        return this.size;
    }
}

class Main {
    public static void main(String[] args) {
        int N;
        int num;
        String cmd = null;
        Scanner sc = new Scanner(System.in);
        Stack st = new Stack();

        N = sc.nextInt();

        for(int i = 0; i < N; i++) {
            cmd = sc.next();
            if(cmd.equals("push")){
                num = sc.nextInt();
                st.push(num);
            } else if(cmd.equals("pop")) {
                System.out.println(st.pop());
            } else if(cmd.equals("top")) {
                System.out.println(st.top());
            } else if(cmd.equals("size")) {
                System.out.println(st.size());
            } else {
                System.out.println(st.empty());
            }
        }
    }
}
